datalad_metalad/extractors/utils.py

```diff
@@ -0,0 +1,18 @@
+"""Common helper functionality for metalad-based and extension extractors """
+
+from datalad.utils import ensure_unicode
+from pathlib import Path
+
+def get_text_from_file(file_path:Path):
+    """Return content of a text file as a string"""
+    # TODO: check that file is text-based
+    file_text = None
+    try:
+        with open(file_path) as f:
+            file_text = ensure_unicode(f.read()).strip()
+            return file_text
+    except FileNotFoundError as e:
+        # TODO: consider returning None in case of exception, depending
+        # on what extractors would expect as default behaviour
+        print((f'The provided file path could not be found: {str(file_path)}'))
```

Comment:
Since we are not handling the error here, we should use a logger with an appropriate logging level instead of print(). Check whether the file path should be reported from `e.filename`.

Doc-string should probably mention that all exceptions from open can be raised, this might be `FileNotFoundError`, `PermissionError`, etc.

Reply:
Thanks!

> Doc-string should probably mention that all exceptions from open can be raised, this might be FileNotFoundError, PermissionError, etc.

Does this imply that I should also just raise any and all possible exceptions in L14, and not just `FileNotFoundError`?

Reply:
Why bother catching, printing (shouldn't be used in general IMHO in a library)?

I would've just used explicit check if file exists is needed and used Path.read_text existing method.

Reply:
What would be good practice to do in such a case when the file does not exist? Return None or raise an exception, or something else?

Reply:
It is hard to answer since it is not clear where such function would be used and why it is needed really, and not just `Path.read_text` or another existing alike... FWIW we have already a "good old" https://github.com/datalad/datalad/blob/HEAD/datalad/utils.py#L2200  which uses our "abominable" `ensure_unicode` which would sweat to help to give you unicode if needed, but it would not do the catch + `print` ;)

Reply:
and if I were to create a function which would return empty string for some reason of "it is ok to not have the file -- then treat it as empty", then would call it `maybe_read_file` or alike ;)

Reply:
Thanks @yarikoptic. I wasn't aware of either `datalad.utils.read_file` or the `Path.read_text` functionality before you mentioned them, so this is a good learning experience for me :)

The background for this helper functionality is described here: https://github.com/datalad/datalad-catalog/issues/39. Basically, I thought it would be good to have a README file content extractor, since such files are fairly ubiquitous in curated datasets of any kind. Then after discussions it morphed into a helper function that can be used by any metalad extractors. If it's just a helper function that gets text from file (and ensures unicode :D), then your suggestions would be sufficient, I think. There are probably other existing/future requirements though, such as reading and distinguishing different types of readme file formats (`txt`, `md`, `rst`, etc) and returning a single or specified format. I've encountered this challenge with the [bids_dataset](https://github.com/datalad/datalad-neuroimaging/pull/104) extractor in `datalad-neuroimaging`, where extracting the text from a `rst` file and loading/dumping it as json results in invalid json. Ideally, IMO, such a helper would take care of all such requirements. But I'm unsure about the best strategy for implementation. Perhaps it's best to take a minimalist approach and just use the existing functionality without building the helper/extractor around it. And then deal with future challenges whenever the future arrives?

Reply:
Avoiding to reinvent the wheel is most often a good decision until you discover that you need wings etc ;-)

Reply:
Ok, so based on the exchange above, @christian-monch @yarikoptic would you say it's fine to drop the PR and use `datalad.utils.read_file` or the `Path.read_text` functionality wherever applicable/useful?

Reply:
I think so, yes